### PR TITLE
Bringing our documentation up to date

### PR DESCRIPTION
**Pull request by RussellSpitzer**
Attempting to enforce consistent Example tags (Helps with Linking)
All links updated to "latest" instead of a specific version
Some Contradictions removed
References to java module removed
Quick Start Simplified

### CODE REVIEW

File: doc/0_quick_start.md [hunk 1]
```diff
@@ -54,28 +53,18 @@ INSERT INTO test.kv(key, value) VALUES ('key2', 2);
...
-Create a `SparkContext`. Substitute `127.0.0.1` with the actual address of your Spark Master
-(or use `"local"` to run in local mode): 
+Run the `spark-shell` with the packages line for your version. To configure
+the default Spark Configuration pass key value pairs with `--conf`
 
-```scala
-val sc = new SparkContext("spark://127.0.0.1:7077", "test", conf)
-```
+    $SPARK_HOME/bin/spark-shell --conf spark.cassandra.connection.host=127.0.0.1 \
+                                --packages datastax:spark-cassandra-connector:1.6.0-s_2.11
```
Comment (bcantoni) [inline doc/0_quick_start.md]: is 2.11 okay here or should this be scala 2.10 version?

File: doc/14_data_frames.md [hunk 2]
```diff
@@ -201,7 +206,7 @@ df.write
 
 ```
 
-###Setting Connector specific options on data frames
+### Setting Connector specific options on data frames
```
Comment (bcantoni) [inline doc/14_data_frames.md]: minor: "data frames" -> "DataFrames"

File: doc/0_quick_start.md (hunk 1 above)
Comment (RussellSpitzer) [inline doc/0_quick_start.md]: That is a good call

**Comment by bcantoni** [comment]
Russ overall looks good with a couple of minor points.

**Pull request merged**